model/span.go

```diff
@@ -28,6 +28,25 @@ var (
 	ErrValidDurationRequired = errors.New("valid duration required")
 )
 
+// Baggage holds the interface for interacting with baggage context propagation
+// implementations.
+type Baggage interface {
+	// Init returns a fresh Baggage primed for usage in a request lifecycle.
+	Init() Baggage
+	// AddHeader adds the provided values to a header designated by key. If key
+	// is not in the whitelist it will be a noop and return false.
+	AddHeader(key string, value ...string) bool
```

Comment:
food for thought, and regardless of "add" vs "replace" design. maybe good to use the word "field" as it isn't always serialized as a header and the word header makes people think about HTTP semantics (ex multi-value type) which can't always be promised anyway (ex in messaging)

Reply:
+1 on use "field". I think `header` feels too coupled with HTTP headers. WRT the multivalues I think it makes sense to follow the existing practice which is multivalue as it is what GRPC metadata and HTTP headers do.

Reply:
Also I am concerned about the `headers` feeling because there are many gotchas around them, notably the `case insensitivity`.